src/Smalot/PdfParser/Encoding.php

```diff
@@ -126,13 +126,24 @@ public function translateChar($dec)
     }
 
     /**
-     * @return string
+     * Returns the name of the encoding class, if available.
      *
-     * @throws \Exception
+     * @return string Returns encoding class name if available or empty string (only prior PHP 7.4).
+     *
+     * @throws \Exception On PHP 7.4+ an exception is thrown if encoding class doesn't exist.
      */
     public function __toString()
     {
-        return $this->getEncodingClass();
+        try {
+            return $this->getEncodingClass();
+        } catch (Exception $e) {
+            // prior to PHP 7.4 toString has to return an empty string.
+            if (version_compare(\PHP_VERSION, '7.4.0', '<')) {
+                return '';
+            } else {
+                throw $e;
+            }
```

Comment:
Optional tiny simplification (but it is actually a coding style matter):

```suggestion
            }
            throw $e;
```